sklearn/decomposition/_base.py

```diff
@@ -57,26 +62,31 @@ def get_precision(self):
         precision : array, shape=(n_features, n_features)
             Estimated precision of data.
         """
+        xp, _ = get_namespace(self.components_)
+
         n_features = self.components_.shape[1]
 
         # handle corner cases first
         if self.n_components_ == 0:
-            return np.eye(n_features) / self.noise_variance_
+            return xp.eye(n_features) / self.noise_variance_
 
-        if np.isclose(self.noise_variance_, 0.0, atol=0.0):
-            return linalg.inv(self.get_covariance())
+        if self.noise_variance_ == 0.0:
```

Comment:
I don't think this is the same as `np.isclose(self.noise_variance_, 0.0, atol=0.0)` because `isclose` has `rtol != 0` by default. However, I am not sure this matters? 

It is a bit annoying that the array API doesn't include a `isclose` :-/

Reply:
I had not realized that. I think we need our own `_isclose` with the same `atol` and `rtol` logic as numpy's.